.circleci/config.yml

```diff
@@ -609,33 +609,53 @@ jobs:
   e2e-tests:
     docker:
       - image: cimg/base:2021.03
+    parallelism: 3
     resource_class: medium+
     steps:
       - checkout
       - copy_secrets_from_lastpass
-      - setup_remote_docker:
-          version: 19.03.13 # https://support.circleci.com/hc/en-us/articles/360050934711
+      - setup_remote_docker
       - run: docker login -u $DOCKER_USER -p $DOCKER_PASS
       - run: |
           cd e2e
-          docker-compose build
+          # --parallel is weird https://github.com/docker/compose/issues/7901 but it seems to work.
+          # `docker compose` (w/o hythen) doesn't work atm "docker: 'compose' is not a docker command"
+          docker-compose build --parallel
+      - run: |
+          cd e2e
           docker-compose run web bin/rails db:drop db:create db:schema:load
           docker-compose run web bin/rails "cl2_back:create_tenant[e2e.front,e2etests_template]"
       - run:
           command: |
             cd e2e
             docker-compose up
           background: true
-      # Wait until build is finished. TODO: Use prebuilt images.
-      - run: sleep 280
+      # Wait until build is finished.
       - run: |
+          sleep 10 # if we run `docker-compose run` immediately, it fails with "network is ambiguous (2 matches found on name)"
           cd e2e
-          docker-compose run --name cypress_run front npm run cypress:run -- --config baseUrl=http://e2e.front:3000
+          # we cannot access localhost:3000, so we call it from the container
+          # https://discuss.circleci.com/t/cannot-communicate-with-docker-container/27253/2
+          docker-compose run front curl --retry 2 --retry-delay 5 --retry-connrefused -v http://e2e.front:3000
+      - run: |
+          cd front
+          TESTFILES=$(circleci tests glob "cypress/e2e/**/*.cy.ts" | circleci tests split  --split-by=timings)
+          COMMA_SEPARATED_TESTFILES=$(echo $TESTFILES | sed 's/ /,/g')
+          echo $COMMA_SEPARATED_TESTFILES
+          cd ../e2e
+          docker-compose run --name cypress_run front npm run cypress:run -- --reporter junit --reporter-options "mochaFile=reports/cypress.xml,toConsole=true" --config baseUrl=http://e2e.front:3000 --spec ${COMMA_SEPARATED_TESTFILES}
+      # We run cypress tests in container, so we need to copy the results out.
+      # ToDo: fix it. Reporting doesn't work atm.
```

Comment:
The Cypress tests run in parallel now, but based on their names, not execution time.